src/sync/app.cpp

```diff
@@ -763,11 +766,14 @@ void App::log_out(std::shared_ptr<SyncUser> user, std::function<void (Optional<A
         util::format("Bearer %1", refresh_token)
     });
 
-    do_request(req, [completion_block, req](Response response) {
+    do_request(req, [&, completion_block = std::move(completion_block)](Response response) {
```

Comment:
I don't think it matters that much.

Reply:
Def don't capture `&` for callbacks that will complete after the current scope returns. If you're trying to capture just `this` you can do [this, completion_block = std::move(completion_block)] to effectively make the lambda a member function.

Reply:
Better yet, you can do `[anchor = shared_from_this(), std::move(completion_block)]` so that you explicitly extend the lifetime of app to the lifetime of the lambda.